lib/node_modules/@stdlib/utils/parse-dsv/lib/main.js

```diff
@@ -0,0 +1,292 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var Buffer = require('buffer').Buffer;
+
+
+// VARIABLES //
+
+var DQUOTE = '"';
+var CR = '\r';
+var LF = '\n';
+var CRLF = '\r\n';
+
+var TOKEN_ENUM = {
+	'DELIM': 0,
+	'DQUOTE': 1,
+	'CR': 2,
+	'LF': 3,
+	'CRLF': 4,
+	'TEXTDATA': 5
+};
+
+var STATE_ENUM = {
+	'INITIAL': 0,
+	'RECORD': 1,
+	'ESCAPED': 2,
+	'NONESCAPED': 3,
+	'TWODQUOTE': 4
+};
+
+
+// MAIN //
+
+/**
+* Returns an a function which incrementally a delimiter separated values resource.
+* @param {Buffer} [DELIM] - the delimiter for the dsv
+* @param {Buffer} [ENCODING] - the encoding of the buffer input
+* @returns {Function} accumulator function
+*
+* @example
+* var Buffer = require( 'buffer' ).Buffer;
+* var parse_dsv = parseDsvStream();
+*
+* var csv = parse_dsv( Buffer.from( '1,2,3' ) );
+* // returns [[ "1", "2", "3" ]]
+*
+* csv = parse_dsv( Buffer.from( '\n' ) );
+* // returns [[ "1", "2", "3" ] ]
+*
+* csv = parse_dsv( Buffer.from('4, 5') );
+* // returns [[ "1", "2", "3" ], [ "4", " 5" ] ]
+*
+* csv = parse_dsv( Buffer.from( ', 6' ) );
+* // returns [ [ "1", "2", "3" ], [ "4", " 5", " 6" ] ]
+*/
+function parseDsvStream( DELIM = ',', ENCODING = 'utf8' ) {
```

Comment:
Eventually, we'll probably opt for a configuration `object` here, rather than keyword arguments.